tests/recognizer-tests.js

```diff
@@ -20,6 +20,14 @@ test("A simple route recognizes", function() {
   equal(router.recognize("/foo/baz"), null);
 });
 
+test("A route with URL-encoded parts recognizes", function() {
+  var handler = {};
+  var router = new RouteRecognizer();
+  router.add([{ path: "/foo bar?baz", handler: handler }]);
```

Comment:
I think url special characters should have to be encoded if they are meant to be matched as part of the path.